### PR TITLE
build: add convenience make targets to install deps and binary

### PR DESCRIPTION
**Pull request by jalateras**
Adde a couple more convenience targets to Makefile.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,6 +4,19 @@ build:
...
 .PHONY: build
 
+# Install the build locally
+install:
+	@go install ./...
+	@echo "==> Installed up ${GOPATH}/bin/up"
+.PHONY: install
+
+# Install the development dependencies
+install_dev_deps:
```
Comment (tj) [inline Makefile]: Not a big deal but I usually do dots for some reason haha maybe `install.deps`

File: Makefile [hunk 2]
```diff
@@ -4,6 +4,19 @@ build:
 	@go generate ./...
 .PHONY: build
 
+# Install the build locally
+install:
+	@go install ./...
+	@echo "==> Installed up ${GOPATH}/bin/up"
+.PHONY: install
```
Comment (tj) [inline Makefile]: Otherwise LGTM! But we should omit bin/bin_assets.go and bindata from this commit, I think it's just the mtime changing

Comment (jalateras) [inline Makefile]: done

File: .gitignore [hunk 3]
```diff
@@ -7,3 +7,4 @@ testing
 up-proxy
 !cmd/up-proxy
 dist
+.idea
```
Comment (hhsnopek) [inline .gitignore]: We should remove this as `.idea` isn't specific to `up(1)` itself, you can add `.idea` to your global `.gitignore`. See https://help.github.com/articles/ignoring-files/ for more details

**Pull request merged**